src/main/java/uk/gov/hmcts/reform/sscs/refdataloader/AppealStatusReferenceDataLoader.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.hmcts.reform.sscs.refdataloader;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.models.refdata.AppealStatus;
+
+@Service
+@Slf4j
+public class AppealStatusReferenceDataLoader {
+
+    public static final String APPEAL_STATUS = "Appeal_Status";
+    public static final String APPEAL_STATUS_ID = "APPEAL_STATUS_ID";
+    public static final String APS_MINOR = "APS_MINOR";
+    public static final String APS_BF_DAYS = "APS_BF_DAYS";
+    public static final String APS_DESC = "APS_DESC";
+    public static final String APS_DORMANT = "APS_DORMANT";
+    public static final String APS_SEARCH = "APS_SEARCH";
+
+    private final Map<Integer, AppealStatus> appealStatusMap = new HashMap<>();
+
+    public void extract(InputStream gaps2RefDataInputStream) throws XMLStreamException {
+
+        if (!appealStatusMap.isEmpty()) {
+            appealStatusMap.clear();
+        }
+
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+
+        AppealStatus appealStatus = null;
+        String tagContent = null;
+
+        try {
+            XMLStreamReader reader =
+                xmlInputFactory.createXMLStreamReader(gaps2RefDataInputStream);
+
+            while (reader.hasNext()) {
+                int event = reader.next();
+
+                switch (event) {
+                    case XMLStreamConstants.START_ELEMENT:
+                        if (APPEAL_STATUS.equals(reader.getLocalName())) {
+                            appealStatus = new AppealStatus();
```

Comment:
If you cannot think of a way to improve this warning in codacy, we can easily ignore that issue in codacy. No need to remove codacy at all.

Reply:
Its good to follow the QA checks recommended by the reforms program. No need to add extra checks.

Reply:
codacy is recommended in the program, Sathish.